OpenRA.Mods.Cnc/FileFormats/BlowfishKeyProvider.cs

```diff
@@ -91,7 +91,7 @@ static void KeyToBigNum(uint[] n, byte[] key, uint len)
 		static uint LenBigNum(uint[] n, uint len)
 		{
 			var i = len - 1;
-			while (n[i] == 0) i--;
+			while (i < uint.MaxValue && n[i] == 0) i--;
 			return i + 1;
```

Comment:
Relying on an underflow followed by an overflow (so that it returns `0` again) feels a bit weird. Considering the `TODO` at the top of the file, maybe we should just change this to an `int` and put the `>= 0` check back.

Reply:
The bits do the same thing for `-1` (`-1 == (int)uint.MaxValue`). I do see your point that it might be confusing to others.

Reply:
`var i = 2U; wile (i > 0 && n[i-1] == 0)`  is more readable to me. if it needs to be fast set `n[0] = !0`.